cmake/compilers/Clang.cmake

```diff
@@ -12,6 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+if(CMAKE_CXX_COMPILER MATCHES "/em\\+\\+(-[a-zA-Z0-9.])?$")
```

Comment:
I recommend building the project using `emcmake`, which will set the `EMSCRIPTEN` variable for you. Then this line could be written simply as:
```
if(EMSCRIPTEN)
```